src/main/java/org/eocqrs/kafka/Consumer.java

```diff
@@ -0,0 +1,33 @@
+/*
+ *  Copyright (c) 2022 Aliaksei Bialiauski, EO-CQRS
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.eocqrs.kafka;
+
+/**
+ * @author Ivan Ivanchuck (l3r8y@duck.com)
+ * @since 0.0.0
+ */
+public interface Consumer<X> {
+
+  void consume(int partition, Data<X> message);
```

Comment:
I believe that this method should return data. But which type of the data? Maybe for now we can go with simple puzzle

Reply:
I have some ideas: we can return Dataized<X> and then turn it in the real data fetched by the consumer

Reply:
I mean turning object into some data

Reply:
@h1alexbel take a look, please at [this](https://github.com/eo-cqrs/eo-kafka/pull/48/commits/448b3c771ea6a9805e5d29cfb780cb24f88cfc77)